src/main/scala/com/github/unchama/seichiassist/subsystems/tradesystems/subsystems/gttosiina/bukkit/listeners/GtToSiinaringo.scala

```diff
@@ -0,0 +1,72 @@
+package com.github.unchama.seichiassist.subsystems.tradesystems.subsystems.gttosiina.bukkit.listeners
+
+import cats.effect.{ConcurrentEffect, IO}
+import com.github.unchama.seichiassist.SeichiAssist
+import com.github.unchama.seichiassist.concurrent.PluginExecutionContexts.onMainThread
+import com.github.unchama.seichiassist.subsystems.gacha.domain.{
+  CanBeSignedAsGachaPrize,
+  GachaPrize
+}
+import com.github.unchama.seichiassist.subsystems.gacha.subsystems.gachaprizefactory.bukkit.StaticGachaPrizeFactory
+import com.github.unchama.seichiassist.subsystems.tradesystems.subsystems.gttosiina.bukkit.traderules.BukkitTrade
+import com.github.unchama.seichiassist.util.InventoryOperations
+import com.github.unchama.util.InventoryUtil.InventoryOps
+import org.bukkit.ChatColor._
+import org.bukkit.Sound
+import org.bukkit.entity.Player
+import org.bukkit.event.inventory.InventoryCloseEvent
+import org.bukkit.event.{EventHandler, Listener}
+import org.bukkit.inventory.ItemStack
+
+class GtToSiinaringo[F[_]: ConcurrentEffect](gachaPrizeTable: Vector[GachaPrize[ItemStack]])(
+  implicit canBeSignedAsGachaPrize: CanBeSignedAsGachaPrize[ItemStack]
+) extends Listener {
+
+  @EventHandler
+  def onGachaRingoEvent(event: InventoryCloseEvent): Unit = {
+    // インベントリをクローズしたのがプレイヤーじゃないとき終了
+    val player = event.getPlayer match {
+      case p: Player => p
+      case _         => return
+    }
+
+    val inventory = event.getInventory
+    val name = player.getName
+
+    // インベントリサイズが4列でない時終了
+    if (inventory.row != 4) return
+
+    if (inventory.getTitle != s"$GOLD${BOLD}椎名林檎と交換したい景品を入れてネ") return
+    // 交換後の情報
+    val tradedInformation =
+      new BukkitTrade(name, gachaPrizeTable).trade(inventory.getContents.toList)
+
+    val tradedAmount = tradedInformation.tradedSuccessResult.map(result => result.amount).sum
```

Comment:
```suggestion
    val totalAmountOfTradeResult = tradedInformation.tradedSuccessResult.map(result => result.amount).sum
```
でよさそうです（提供したアイテムの数なのか受け取ったアイテムの数なのかが曖昧）